azurerm/internal/services/compute/linux_virtual_machine_resource.go

```diff
@@ -816,6 +815,23 @@ func resourceLinuxVirtualMachineUpdate(d *schema.ResourceData, meta interface{})
 		}
 	}
 
+	if d.HasChange("allow_extension_operations") {
+		provisionVMAgent := d.Get("provision_vm_agent").(bool)
+		allowExtensionOperations := d.Get("allow_extension_operations").(bool)
+
+		if !provisionVMAgent && allowExtensionOperations {
+			return fmt.Errorf("`allow_extension_operations` cannot be set to `true` when `provision_vm_agent` is set to `false`")
```

Comment:
This is not correct. provision_vm_agent affects the create process of a virtual machine.
Assume you did not install the agent (provision_vm_agent=false), then you can still install the agent manually

https://docs.microsoft.com/en-us/azure/virtual-machines/troubleshooting/install-vm-agent-offline

As far as I know, the manual installation does not affect provision_vm_agent value, but you can still use allow_extension_operations